### DISCUSSION TITLE
Critical dependency: the request of a dependency is an expression

### DISCUSSION THREAD

**Discussion by ollebergkvist** [apify/got-scraping#111] [Q&A]
I'm trying to use this package within a route handler (Next.js v.13 app directory (latest version), but I'm getting:
Import trace for requested module:
./node_modules/header-generator/node_modules/browserslist/node.js
./node_modules/header-generator/node_modules/browserslist/index.js
./node_modules/header-generator/utils.js
./node_modules/header-generator/header-generator.js
./node_modules/header-generator/index.js
./node_modules/got-scraping/dist/index.js
./src/app/api/hltv/team/[id]/route.ts

./node_modules/keyv/src/index.js
Critical dependency: the request of a dependency is an expression

Wondering if anyone knows how I can make it work in this env?

**Comment by B4nan** [comment]
Sounds like this? https://github.com/i18next/next-i18next/issues/1545#issuecomment-1236260356

You can't use this package from frontend context.

**Reply by ollebergkvist** [reply]
@B4nan 

https://nextjs.org/docs/app/building-your-application/routing/route-handlers

I'm using the package server side in this case, so it's not run on the client.

**Reply by Hamza5** [reply]
@ollebergkvist Check this solution: https://github.com/apify/got-scraping/discussions/143#discussioncomment-11129372
